### PR TITLE
refactor(ld-input): better event handling

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

This change defines events explicitly, overriding native events, fixing the onChange handler. Works in docs page, but doesn't work in React yet. When you change the event name from `change` to something that is not already a name of native event, the handler works in React. Not there yet...

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] unit tests
- [x] manual tests in React App

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-input/ld-input.tsx [hunk 1]
```diff
@@ -133,24 +133,9 @@ export class LdInput implements InnerFocusable, ClonesAttributes {
...
-  /**
-   * Emitted when the input receives focus.
-   */
-  @Event() focus: EventEmitter<FocusEvent>
-
-  /**
-   * Emitted when the input value changes.
-   */
-  @Event() input: EventEmitter<string>
+  @Event() ldchange: EventEmitter<string>
```
Comment (borisdiakur) [inline src/liquid/components/ld-input/ld-input.tsx:138]: Do I understand it correctly that this is an additional event to the `change` event, which is still available (though not documented anymore), except for when using React? If that is the case, we should probably document this somewhere.

File: src/liquid/components/ld-input/test/ld-input.spec.ts [hunk 2]
```diff
@@ -61,52 +61,47 @@ describe('ld-input', () => {
     expect(ldInput).toMatchSnapshot()
   })
 
-  it('emits focus and blur event', async () => {
+  xit('emits focus and blur event', async () => {
```
Comment (borisdiakur) [inline src/liquid/components/ld-input/test/ld-input.spec.ts:64]: Let's leave a comment why we `xit`ed this one.

File: src/liquid/components/ld-input/ld-input.tsx (hunk 1 above)
Comment (renet) [reply src/liquid/components/ld-input/ld-input.tsx:138]: Yes, exactly. Maybe we should add a general section about native events to the docs, making clear that there's a difference between composed and non-composed native events and that we add additional custom events for some non-composed events. I would do that in a separate PR that we should also use to define a global convention for events in Liquid Oxygen and then refactor other components accordingly.

File: src/liquid/components/ld-input/test/ld-input.spec.ts (hunk 2 above)
Comment (renet) [reply src/liquid/components/ld-input/test/ld-input.spec.ts:64]:
```suggestion
  // TODO: Uncomment, as soon as Stencil's JSDom implementation
  // supports bubbling of composed events into the light DOM.
  xit('emits focus and blur event', async () => {
```

**Review by renet — APPROVED** [review]

**Pull request merged**